snuba/utils/streams/batching.py

```diff
@@ -165,9 +142,6 @@ def run(self) -> None:
     def _run_once(self) -> None:
         self._flush()
 
-        if self.producer:
-            self.producer.poll(0.0)
```

Comment:
This was moved to `KafkaConsumerWithCommitLog.poll`. It is required to ensure the `on_delivery` callback of the `Producer` is called so that you can be notified if you are not able to deliver messages.

Reply:
> And why it was here instead of being upon commit ?

Also to answer this more specifically, it's here so the callback queue is flushed more frequently, hopefully leading to less memory consumption and earlier problem detection.